lib/evm_database_ops.rb

```diff
@@ -119,15 +117,40 @@ def self.restore(db_opts, connect_opts = {})
         backup_folder = action == :dump ? "db_dump" : "db_backup"
         uri = File.join(connect_opts[:uri], backup_folder, connect_opts[:remote_file_name])
       end
+    else
+      uri = db_opts[:local_file]
+
+      # HACK(ish): Since everything is flowing through MiqFileStorage so that
+      # we can handle the file splitting logic all in one place, this means
+      # that local dumps/backups also need to now have a "uri" in the
+      # connection opts so that MiqLocalMountSession will be used.
+      #
+      # This just puts the bare minimum necessary for URI.parse to recognize
+      # the :uri option as "file" scheme, and allows MiqFileStorage to then
+      # instanciate MiqLocalMountSession below in the `.with_interface_class`
+      # method.
+      connect_opts[:uri] = "file://"
+    end
 
-      session = MiqGenericMountSession.new_session(connect_opts)
-      db_opts[:local_file] = session.uri_to_local_path(uri)
+    MiqFileStorage.with_interface_class(connect_opts) do |file_storage|
+      send_args = [uri, db_opts[:byte_count]].compact
+
+      # Note:  `input_path` will always be a fifo stream (input coming from
+      # PostgresAdmin, and the output going to the `uri`), since we want to
+      # maintain the same interface for all backup types.  This means that
+      # `uri` will always be the final destination, but `input_path` below will
+      # be an intermediary fifo that will take the input from `pg_dump`,
+      # `pg_restore`, or `pg_basebackup`, and stream the results (in ruby) it
+      # to whatever endpoint it might be the resulting `uri`.
```

Comment:
> and stream the results (in ruby) it to whatever endpoint it might be the resulting `uri`.

This end bit of the sentence here just doesn't make sense.